### PR TITLE
Internal, automatic use of get direct.

### PR DESCRIPTION
**Pull request by scottf**

### CODE REVIEW

File: src/main/java/io/nats/client/api/MessageInfo.java [hunk 1]
```diff
@@ -83,12 +127,20 @@ public Headers getHeaders() {
...
+        return lastSeq;
+    }
+
     @Override
     public String toString() {
         return "MessageInfo{" +
                 "subject='" + subject + '\'' +
                 ", seq=" + seq +
-                ", data='" + data + '\'' +
+                ", data='" + (data == null ? "null" : new String(data, UTF_8)) + '\'' +
```
Comment (arondi) [inline src/main/java/io/nats/client/api/MessageInfo.java:143]: also need the rest of the direct header data

File: src/main/java/io/nats/client/impl/NatsJetStreamImplBase.java [hunk 2]
```diff
@@ -23,11 +23,22 @@
 import java.io.IOException;
 import java.time.Duration;
 import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
 
 import static io.nats.client.support.ApiConstants.SUBJECT;
 
 class NatsJetStreamImplBase implements NatsJetStreamConstants {
 
+    static class CachedStreamInfo {
```
Comment (arondi) [inline src/main/java/io/nats/client/impl/NatsJetStreamImplBase.java:32]: add dev note here

File: src/main/java/io/nats/client/impl/NatsJetStreamManagement.java [hunk 3]
```diff
@@ -62,7 +63,7 @@ private StreamInfo addOrUpdateStream(StreamConfiguration config, String template
 
         String subj = String.format(template, streamName);
         Message resp = makeRequestResponseRequired(subj, config.toJson().getBytes(StandardCharsets.UTF_8), jso.getRequestTimeout());
-        return new StreamInfo(resp).throwOnHasError();
+        return createAndCacheStreamInfo(streamName, resp);
```
Comment (arondi) [inline src/main/java/io/nats/client/impl/NatsJetStreamManagement.java:66]: what happens on an error?

File: src/main/java/io/nats/client/support/NatsJetStreamClientError.java [hunk 4]
```diff
@@ -5,6 +5,7 @@ public class NatsJetStreamClientError {
     private static final int KIND_ILLEGAL_STATE = 1;
     private static final String SUB = "SUB";
     private static final String SO = "SO";
+    private static final String JSM = "JSM";
```
Comment (arondi) [inline src/main/java/io/nats/client/support/NatsJetStreamClientError.java:8]: this isn't needed.

File: src/main/java/io/nats/client/support/NatsJetStreamConstants.java [hunk 5]
```diff
@@ -68,6 +68,9 @@ public interface NatsJetStreamConstants {
     // JSAPI_DIRECT_GET is the endpoint to directly get a message.
     String JSAPI_DIRECT_GET = "DIRECT.GET.%s";
 
+    // JSAPI_DIRECT_GET is the shortcut endpoint to directly get a last by subject message
```
Comment (arondi) [inline src/main/java/io/nats/client/support/NatsJetStreamConstants.java:71]: better comment - is shortcut the right word

**Review by arondi — CHANGES_REQUESTED** [review]

File: src/main/java/io/nats/client/api/MessageInfo.java (hunk 1 above)
Comment (scottf) [reply src/main/java/io/nats/client/api/MessageInfo.java:143]: fixed

File: src/main/java/io/nats/client/impl/NatsJetStreamManagement.java (hunk 3 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsJetStreamManagement.java:66]: renamed the method to indicate it throws on error

File: src/main/java/io/nats/client/impl/NatsJetStreamImplBase.java (hunk 2 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/NatsJetStreamImplBase.java:32]: done

File: src/main/java/io/nats/client/support/NatsJetStreamClientError.java (hunk 4 above)
Comment (scottf) [reply src/main/java/io/nats/client/support/NatsJetStreamClientError.java:8]: removed

File: src/main/java/io/nats/client/support/NatsJetStreamConstants.java (hunk 5 above)
Comment (scottf) [reply src/main/java/io/nats/client/support/NatsJetStreamConstants.java:71]: done

**Review by arondi — APPROVED** [review]
LGTM

**Pull request merged**